src/release/breaking-changes/flutter-gradle-plugin-apply.md

```diff
@@ -0,0 +1,196 @@
+
+To learn about advantages the new Plugin DSL syntax has over the legacy `apply`
+script syntax, see [Gradle docs][plugins block].
+
+Migrating the app ecosystem to use the new approach will also make it easier for
+Flutter team to develop Flutter's Gradle plugins, and to enable exciting new
+features in the future, such as using Kotlin instead of Groovy in Gradle
+buildscripts.
+
+## Migrate
+
+### android/settings.gradle
+
+Replace the contents of `<app-src>/android/settings.gradle` with:
+
+```gradle
+pluginManagement {
+    def flutterSdkPath = {
+        def properties = new Properties()
+        file("local.properties").withInputStream { properties.load(it) }
+        def flutterSdkPath = properties.getProperty("flutter.sdk")
+        assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
+        return flutterSdkPath
+    }
+    settings.ext.flutterSdkPath = flutterSdkPath()
+
+    includeBuild("${settings.ext.flutterSdkPath}/packages/flutter_tools/gradle")
+
+    repositories {
+        google()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+}
+
+plugins {
+    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
+    id "com.android.application" version "7.3.0" apply false
+    id "org.jetbrains.kotlin.android" version "1.7.10" apply false
+}
+
+include ":app"
+```
+
+### android/build.gradle
+
+Remove the whole `buildscript` block from `<app-src/android/build.gradle`:
+
+```diff
+-buildscript {
+-    ext.kotlin_version = '1.7.10'
+-    repositories {
+-        google()
+-        mavenCentral()
+-    }
+-
+-    dependencies {
+-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+-    }
+-}
+```
+
+Here's how that file will likely end up:
+
+```gradle
+allprojects {
+    repositories {
+        google()
+        mavenCentral()
+    }
+}
+
+rootProject.buildDir = '../build'
+subprojects {
+    project.buildDir = "${rootProject.buildDir}/${project.name}"
+}
+subprojects {
+    project.evaluationDependsOn(':app')
+}
+
+tasks.register("clean", Delete) {
+    delete rootProject.buildDir
+}
+```
+
+### android/app/build.gradle
+
+Most changes have to be made in the `<app-src>/android/app/build.gradle`. First,
+remove these 2 chunks of code that use the legacy imperative apply method:
+
+```diff
+-def flutterRoot = localProperties.getProperty('flutter.sdk')
+-if (flutterRoot == null) {
+-    throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
+-}
+```
+
+```diff
+-apply plugin: 'com.android.application'
+-apply plugin: 'kotlin-android'
+-apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
+```
+
+Now apply the plugins again, but this time using the Plugin DSL syntax. At the
+very top of your file, add:
+
+```diff
++plugins {
++    id "com.android.application"
++    id "kotlin-android"
++    id "dev.flutter.flutter-gradle-plugin"
++}
+```
+
+## Examples
```

Comment:
Before the example I think we should have a section that gives a user a way to validate they migrated successfully. 

Perhaps something like 
```
### Validation 

Execute  `flutter run apk` and confirm your app builds and launches. 

## Examples 
```

Reply:
done in b7f022f93e7798f3b5eec6b69027372d56ac4762